templates/master/00-master/baremetal/files/baremetal-haproxy.yaml

```diff
@@ -41,6 +41,9 @@ contents:
             old_pids=$(pidof haproxy)
             if [ -n "$old_pids" ]; then
                 /usr/sbin/haproxy -W -db -f /etc/haproxy/haproxy.cfg  -p /var/lib/haproxy/run/haproxy.pid -x /var/lib/haproxy/run/haproxy.sock -sf $old_pids &
+                #There seems to be some cases where HAProxy doesn't drain properly.
+                #To handle that case, SIGTERM signal being sent to old HAProxy processes which haven't terminated.
+                sleep 60  && kill $old_pids &
```

Comment:
What happens if during that 60 seconds another process gets assigned the same pid? Should we re-run pidof before doing this?

Reply:
1. From the tests  I did in dev-scripts environment, No process was deleted after calling 'haproxy -sf' option, I guess it because of the timeout values we use in Haproxy cfg.

2. If we'll re-run pidof, we'll get also the new haproxy process (the one we don't want to delete), 
so we should run intersect logic between the pidof lists. 

3. Based on the tests on dev-scripts, I don't think we should add this extra logic, let me know if you still think this logic is required.

Reply:
If -sf isn't doing anything, should we just remove it and kill the process immediately? That would eliminate any possibility of a race without any extra logic. It would also remove an unnecessary 60 second delay in stopping the old processes. Or does -sf allow it to to stop some connections?

Reply:
I think that -sf does something ( stops some connections), so to be on the safe side, I suggest to keep the long timeout values and add the kill after timeout (default 120 seconds).
Also to verify that we kill a HAProxy process and not another process that gets assigned the same pid.